### PR TITLE
roles yml changes for security-analytics plugin

### PR DESCRIPTION
**Pull request by raj-chak**
Signed-off-by: Raj Chakravarthi <raj@icedome.ca>

### Description
Added entries in roles.yml for cluster permissions to security-analytics plugin end points
Added default roles configuration required for `security-analytics` plugin

### Issues Resolved
https://github.com/opensearch-project/security-analytics/issues/50

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: config/roles.yml [hunk 1]
```diff
@@ -255,3 +255,26 @@ point_in_time_full_access:
         - '*'
       allowed_actions:
         - 'manage_point_in_time'
+
+# Allows users to see security analytics detectors and others
+detectors_read_access:
```
Comment (peternied) [inline config/roles.yml:260]: There already is a role called `alerting_read_access` | `alerting_full_access`, seem like they could be confusing for users trying to pick between the two, are there other names that might be more closely related to your plugin?

File: config/roles.yml [hunk 2]
```diff
@@ -255,3 +255,26 @@ point_in_time_full_access:
...
+  reserved: true
+  cluster_permissions:
+    - 'cluster:admin/opendistro/securityanalytics/alerts/get'
```
Comment (peternied) [inline config/roles.yml:263]: This looks like a new plugin, can we swap the `opendistro` nomenclature for `opensearch`?

File: config/roles.yml [hunk 3]
```diff
@@ -255,3 +255,26 @@ point_in_time_full_access:
...
+    - 'cluster:admin/opendistro/securityanalytics/rule/get'
+    - 'cluster:admin/opendistro/securityanalytics/detector/get'
+    - 'cluster:admin/opendistro/securityanalytics/rule/search'
+    - 'cluster:admin/opendistro/securityanalytics/detector/search'
+
+# Allows users to use all security analytics functionality
+detectors_full_access:
+  reserved: true
+  cluster_permissions:
+    - 'cluster:admin/opendistro/securityanalytics/mapping/*'
```
Comment (peternied) [inline config/roles.yml:276]: Please sort all permissions in alphabetical order to make them easier to compare between read/full roles

**Review by peternied — CHANGES_REQUESTED** [review]
Thanks for creating this pull request, can you please fill in the pull request description for this change?

Biggest unknown for me is where the design usage of these permissions, would love to get more details on this.  Please link associated issues/documents about this plugin.

File: config/roles.yml (hunk 1 above)
Comment (raj-chak) [reply config/roles.yml:260]: the two changes start with the word "detectors". That has been the naming convention

File: config/roles.yml (hunk 2 above)
Comment (raj-chak) [reply config/roles.yml]: swapped to opensearch

File: config/roles.yml (hunk 3 above)
Comment (raj-chak) [reply config/roles.yml]: done

File: config/roles.yml (hunk 1 above)
Comment (getsaurabh02) [reply config/roles.yml:260]: Should we call them ``security_analytics_read_access`` and so on?

File: config/roles.yml [hunk 4]
```diff
@@ -255,3 +255,31 @@ point_in_time_full_access:
...
+  cluster_permissions:
+    - 'cluster:admin/opensearch/securityanalytics/alerts/get'
+    - 'cluster:admin/opensearch/securityanalytics/detector/get'
+    - 'cluster:admin/opensearch/securityanalytics/detector/search'
+    - 'cluster:admin/opensearch/securityanalytics/findings/get'
+    - 'cluster:admin/opensearch/securityanalytics/mapping/get'
+    - 'cluster:admin/opensearch/securityanalytics/mapping/view/get'
+    - 'cluster:admin/opensearch/securityanalytics/rule/get'
+    - 'cluster:admin/opensearch/securityanalytics/rule/search'
+
```
Comment (getsaurabh02) [inline config/roles.yml:271]: Lets also add ``security_analytics_ack_alerts`` for the security operators to be able to acknowledge alerts with cluster permission as:

```
cluster_permissions:
    - 'cluster:admin/opendistro/securityanalytics/alerts/*'
```

File: config/roles.yml [hunk 5]
```diff
@@ -255,3 +255,31 @@ point_in_time_full_access:
...
+    - 'cluster:admin/opensearch/securityanalytics/alerts/*'
+    - 'cluster:admin/opensearch/securityanalytics/detector/*'
+    - 'cluster:admin/opensearch/securityanalytics/findings/*'
+    - 'cluster:admin/opensearch/securityanalytics/mapping/*'
+    - 'cluster:admin/opensearch/securityanalytics/rule/*'
+  index_permissions:
+    - index_patterns:
+        - '*'
+      allowed_actions:
+        - 'indices:admin/mapping/put'
```
Comment (getsaurabh02) [inline config/roles.yml:285]: How about  ``indices:admin/mappings/get`` ? Also are there more permissions such as for aliases that might be needed?

**Review by getsaurabh02 — CHANGES_REQUESTED** [review]

File: config/roles.yml (hunk 5 above)
Comment (raj-chak) [reply config/roles.yml:285]: done and tested.

File: config/roles.yml (hunk 1 above)
Comment (raj-chak) [reply config/roles.yml]: done

File: config/roles.yml (hunk 4 above)
Comment (raj-chak) [reply config/roles.yml:271]: done

**Comment by peternied** [comment]
> ❌ CI / backward-compatibility (pull_request) Failing after 5m

This looks unrelated to this change, need to see if there is a new failure from main

**Review by DarshitChanpura — APPROVED** [review]

**Review by peternied — APPROVED** [review]
Thanks for making these updates, merging!

**Comment by peternied** [comment]
Note; the BWC issue is on main, merging this despite BWC failures since they were not impacted

**Pull request merged**